### ISSUE TITLE
Better one-handed use: double-tap to submit choices in quests

### ISSUE THREAD

**Issue by ash211** [streetcomplete/StreetComplete#4464]
**Use case**
Sometimes I'm walking the dog in one hand and using StreetComplete in the other hand, and have some frustration using StreetComplete with one hand.  Specifically, on screens like these two (from the README), tapping the choice with the left hand is fine but reaching the "OK" button location with left hand is a struggle (on a moderately-large size phone).

![687474703a2f2f7777772e776573746e6f72646f73742e64652f737472656574636f6d706c6574652f70686f6e6553637265656e73686f74732f73637265656e73686f7430362e706e67](https://user-images.githubusercontent.com/357170/193975285-cf7146e8-e479-448d-9cf9-92b8f69ed57b.png)
![687474703a2f2f7777772e776573746e6f72646f73742e64652f737472656574636f6d706c6574652f70686f6e6553637265656e73686f74732f73637265656e73686f7430382e706e67](https://user-images.githubusercontent.com/357170/193975289-21065904-eee5-4b7c-a77c-2eb58946aece.png)

**Proposed Solution**
Support submitting in choices like these by double-tapping on the option.  To be specific, double-tapping on the "Concrete" choice should be equivalent to single-tapping on concrete followed by single-tapping on "OK".  Same with "Warehouse" in the second screenshot.

I've seen this double-tapping pattern elsewhere in the Android ecosystem and think it would be useful for StreetComplete too.

Thanks for the app, I love using it to contribute to OSM!

**Comment by mnalis** [comment]
Relatedly, being right handed, I have same problem with taking pictures for Notes (which I use a lot), as those are too far to the left to reach with one hand on moderately sized phone:
![small_Screenshot_20221005_111751_de westnordost streetcomplete](https://user-images.githubusercontent.com/156656/194026308-346ce725-1e0a-47ed-ae47-2cabbb5a1e2a.jpg)

It bothered me enough that I've worked around that by reordering UI elements in my fork, but perhaps a **global StreetComplete preference for "right-handed" vs. "left-handed"**  and reordering position of such few key buttons would be alternative which could solve all those problems without having to introduce such gestures (e.g. doubletapping is prone to problems when driving in vehicles on less-then-perfect road surfaces)

Another workaround I sometimes use (e.g. when needed to use full-size keyboard one handed) is to reduce my screen size by using android `One-Handed mode` feature (on my phone it is mapped to holding then dragging from center navigation circle to the right) which reduces screen size by about 1/3, making all screen elements reachable by one hand. But of course it is very annoying, not only because one has to enter and exit such mode frequently, but also as smaller screen means smaller icons, harder to hit correctly keys etc.

**Comment by westnordost** [comment]
Hmm, I'd say it is kind of impossible to design the app in a way that all important clickable elements are somewhere within the area reachable with the thumb on the lower right (or left, for left-handed people) third of the screen. It for example already does not work if you need to pull up the surface or building selection sheet and select another building type.

Your suggestion, to make double-tap confirm the selection, would be something that is very intransparent to the user as it is not explained anywhere and can happen just as well accidentally.

I know, it is probably not that helpful, but I remember a time, not so long ago, where smartphones in general were small enough that you could operate them with one hand+thumb no problem. I'd say, phones with a screen of up to 5.6 inches. My current phone (the smallest I could get, Samsung S10e) has 5.8" and I can reach everything except the upper-left third. There are some niche phones with very small screens around that may be fit as dog-walking phones, such as the Jelly2 or the Cubot phones. 

Samsung specifically (I think) has this feature called ["one handed mode"](https://www.samsung.com/in/support/mobile-devices/samsung-galaxy-smartphones-one-hand-mode/) with which you can effectively make the screen smaller. Maybe other vendors have this feature too, in case you have no current Samsung software.

**Comment by mnalis** [comment]
Well I have Huawei P30 Pro which has larger 6.47" screen, and as right-handed person I can use all _**common**_ functionalities of upstream SC, except:

- **rotating the screen** (pending https://github.com/streetcomplete/StreetComplete/pull/3954). That can be worked around by spending mental credits, but finishing that as software PR would be much better.
- the before-mentioned **"take image" icon in Notes** (https://github.com/streetcomplete/StreetComplete/issues/4464#issuecomment-1268184975) which I've already worked around in my fork (but that may be just me - I'm quite Notes-heavy user; although that might reduce somewhat now with ability to add POIs from SC directly)
- quests which require lots of **full text keyboard** (like place name) - Some parts of the keyboard are ok, but some or not - there is no help for that one but (as mentioned) forcing phone to reduced screen size (AKA "one handed mode", which BTW also works on my Huawei with EMUI 12, but as noted before is annoying to use), but it is **_luckily relatively rare_** compared to all other quests. Or maybe some custom keyboard might exist which uses only part of the screen? 

Sure, not 100% of the features can be made one-hand reachable; but if the **_majority of most common ones could be_** (as I did in my fork; it works - but is hardcoded for right-hand people), that would make it much more likely to be used in such way.  

<details>

<summary>smaller screens, dogs and other workarounds (expand)</summary>

The problem with smaller phones (which used to be common), as well as reduced screen size in _one handed mode_, is that as one ages they become too small to read and click accurately. 

As for the dogs, one can also train them not to pull on the leash but stay closer to owner so (s)he can use both hands, but that is also a non-typical solution to software problem :smiley: -- and there are other reasons why one might not use other hand (like reassuring their SO that they **_are_** _as important_ as [OSM mapping](https://wiki.openstreetmap.org/wiki/OSM_purity_self-test) :rofl: ) 

</details>


---

TL;DR: it is **fine if one has to use both hand every 10 minutes** to do some special thing, it is still very usable with one hand. If one has to use both hands **every 30 seconds, then it is not** (which I guess is what is happening to **_left-handed people_** trying to use SC with one hand).

**Comment by westnordost** [comment]
Anyway, I'll close this, see https://github.com/streetcomplete/StreetComplete/issues/4464#issuecomment-1268287169
